### ISSUE TITLE
[BUG] No CSS Styling in CI/CD

### ISSUE THREAD

**Issue by snazzyj** [microsoft/playwright#28742]
### System info
- Playwright Version: [v1.36.1]
- Operating System: [Windows 11]
- Browser: [Chromium]
- Other info: Vue2, Vuetify2 for Frontend frameworks.  Using webpack as our bundler

### Source code

Unfortunately, best I can do is a screenshot showing our Trace report timeline.  It does not seem that any CSS styling is being applied.  This does not happen on any of our local dev environments and only in the CI/CD pipeline.  I might be able to link our `vue.config.ts` file and some other configuration files but I'm really not sure what is needed.

![image](https://github.com/microsoft/playwright/assets/48457459/055b01a6-3353-47d6-9d02-15c0aeb2ac07)

**Steps**
- NONE

**Expected**

Screenshot shows interface page loading with all CSS styling applied.

![image](https://github.com/microsoft/playwright/assets/48457459/4ff90470-f410-4f39-96e0-222a1f979047)

**Actual**

Screenshot shows interface page loading with NO CSS styling applied.
![image](https://github.com/microsoft/playwright/assets/48457459/055b01a6-3353-47d6-9d02-15c0aeb2ac07)

**Comment by yury-s** [comment]
Can yo share a test along with the page where it's happening? Or at least a trace file with broken css?

**Comment by yury-s** [comment]
Actually, screenshots in the trace also show the page without css styling, so it looks like on CI you build the site differently and it doesn't apply the css.

**Comment by snazzyj** [comment]
@yury-s Apologies for my lack of knowledge on how CI/CD works under the hood.  Why would this happen on a test CI but not our builds / deploys? 

We have the PW config setup to run a `npm run all` command when starting up the server in CI.  This in turns runs something like:

```
scripts: {
    "all": "concurrently -k -p \"[{name}]\" -n \"FRONTEND,API,WORKER\" -c \"red,cyan,blue\" \"npm run serve\" \"npm run api\"  \"npm run worker\"",
```

Could it in theory be the `npm run serve` command being used that is causing this to not build + apply CSS properly?  As stated, this doesn't happen on local where I'm able to run the same YAML script command of "npm run test"

**Comment by yury-s** [comment]
> Why would this happen on a test CI but not our builds / deploys?

CI machines are often times less powerful (e.g. on GitHub Actions standard one would have 2 CPU cores) than local ones which may affect timing of the operations running in parallel. I see that you run several commands in parallel in the "all" script you mentioned and that may behave differently. My hunch is it might happen that the tests start running before the server is fully up, if that's the case you can use [webServer](https://playwright.dev/docs/api/class-testconfig#test-config-web-server) config property instead to make playwright wait.

> Could it in theory be the npm run serve command being used that is causing this to not build + apply CSS properly? As stated, this doesn't happen on local where I'm able to run the same YAML script command of "npm run test"

I don't what that command does in you project, so probably it could. Maybe the build is not complete yet by the time the tests are running.

In any case, this is a problem with your project setup rather than playwright, so I'm closing this report. If you think there is a bug in playwright, feel free to open another issue and attach a repro that we could run locally to reproduce it.

**Comment by snazzyj** [comment]
@yuri-s Thanks for your insight.  Got some ideas between this thread and the discord :)
